src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,71 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int AMOUNT_INDEX = 1;
+    private static final int TYPE_INDEX = 0;
+    private static final String SUPPLY = "supply";
+    private static final String RESULT = "result";
+    private static final String BUY = "buy";
+    private static final String COMMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String text = createReport(fromFileName);
+        writeReportToFile(toFileName, text);
+    }
+
+    private int[] readFile(String fromFile) {
+        int[] result = new int[3];
+        int supplyAmount = 0;
+        int buyAmount = 0;
+        int resultAmount;
+        try (BufferedReader reader
+                     = new BufferedReader(new FileReader(fromFile))) {
+            String data = reader.readLine();
+            while (data != null) {
+                String[] line = data.split(COMMA);
+                if (line[TYPE_INDEX].equals(SUPPLY)) {
+                    supplyAmount += Integer.parseInt(line[AMOUNT_INDEX]);
+                } else {
+                    buyAmount += Integer.parseInt(line[AMOUNT_INDEX]);
+                }
+                data = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file" + fromFile, e);
+        }
+        resultAmount = supplyAmount - buyAmount;
+        result[0] = supplyAmount;
+        result[1] = buyAmount;
+        result[2] = resultAmount;
+        return result;
+    }
+
+    private String createReport(String fromFile) {
+        int[] data = readFile(fromFile);
```

Comment:
it is better to call all private methods in a public method and pass the result as an input parameter to the next processing method